### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by popovychyb**

### CODE REVIEW

File: src/main/java/core/basesyntax/Pair.java [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Pair.java]:
```suggestion
        Pair<T, S> pair = (Pair) o;
```

File: src/main/java/core/basesyntax/Pair.java [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+        return Objects.equals(first, pair.first)
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Pair.java]: try to do it without Objects class and same hashCode()
Hint: check yourself with checklist before sending your solution

File: src/main/java/core/basesyntax/Pair.java (hunk 1 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/Pair.java]: Done. Thank you!

File: src/main/java/core/basesyntax/Pair.java (hunk 2 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/Pair.java]: Done. Thank you!

File: src/main/java/core/basesyntax/Pair.java [hunk 3]
```diff
@@ -1,5 +1,47 @@
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Pair<T, S> pair = (Pair) o;
+
+        if (first != null ? !first.equals(pair.first) : pair.first != null) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Pair.java]: try simplify this method:
```
return (pair.getFirst() == this.first
                || (pair.getFirst() != null && pair.getFirst().equals(this.first)))
                && (pair.getSecond() == this.second
                || (pair.getSecond() != null && pair.getSecond().equals(this.second)));
```

File: src/main/java/core/basesyntax/Pair.java [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+    private T first;
+    private S second;
 
+    private Pair(T first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T, S> Pair of(T first, S second) {
+        return new Pair(first, second);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Pair.java]:
```suggestion
        return new Pair<T, S>(first, second);
```

**Review by RomanDubovskyi — COMMENTED** [review]
almost there

File: src/main/java/core/basesyntax/Pair.java (hunk 3 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/Pair.java]: Done. Thank you for suggestions!

File: src/main/java/core/basesyntax/Pair.java (hunk 4 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/Pair.java]: Done. Thank you!

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**